src/command/dev/protocol/leader.rs

```diff
@@ -60,6 +60,7 @@ impl LeaderSession {
     /// Ok(Some(Self)) when successfully initiated
     /// Ok(None) when a LeaderSession already exists for that address
     /// Err(RoverError) when something went wrong.
+    #[allow(clippy::too_many_arguments)]
```

Comment:
I'm cheating here, but fixing this will require some refactoring that might complicate this PR. Happy to do it, but wondering if it's worth it given the potential rover dev rewrite?